src/utils.ts

```diff
@@ -136,18 +139,32 @@ export function getNpmConfig(runnerConfig: NpmConfigContainer) {
       runnerConfig.npm.legacyPeerDeps !== false,
   };
 
-  // As npm config accepts only key-value pairs, we do the translation
+  // As npm config accepts only key-value pairs, we do the translation.
   if (runnerConfig.npm.registries) {
-    for (const sr of runnerConfig.npm.registries) {
-      if (sr.scope) {
-        cfg[`${sr.scope}:registry`] = sr.url;
+    for (const r of runnerConfig.npm.registries) {
+      if (r.scope) {
+        cfg[`${r.scope}:registry`] = r.url;
       } else {
-        cfg.registry = sr.url;
+        cfg.registry = r.url;
       }
 
-      if (sr.authToken) {
-        const field = getRegistryAuthConfigField(sr.url);
-        cfg[field] = sr.authToken;
+      // Configures npm auth fields by prefixing a scoped URI.
+      // For more info, check the npm doc https://docs.npmjs.com/cli/v10/configuring-npm/npmrc#auth-related-configuration
+      const uri = getRegistryURI(r.url);
+      if (r.authToken) {
+        cfg[`${uri}_authToken`] = r.authToken;
+      }
+      if (r.auth) {
+        cfg[`${uri}_auth`] = r.auth;
+      }
+      if (r.username) {
+        cfg[`${uri}username`] = r.username;
+      }
+      if (r.password) {
+        cfg[`${uri}_password`] = r.password;
+      }
+      if (r.email) {
+        cfg[`${uri}email`] = r.email;
```

Comment:
Missing `:`. See https://docs.npmjs.com/cli/v10/configuring-npm/npmrc#auth-related-configuration.

Reply:
The `:` is added in `getRegistryURI`, though it's a bit confusing. But I read the docs and it states 

> the scope may look like `//registry.npmjs.org/:`

So I put it into `getRegistryURI`. I can move it back.

Reply:
Oh my bad, didn't see!
If you want to keep it in `getRegistryURI()`, then I'd recommend renaming that method to `getRegistryURIFragment`. Same with the local var `uri` -> `uriFragment`.
That way it's clearer that this is no mere URI.

Reply:
>  Same with the local var uri -> uriFragment.

Still missing 👁️